### PR TITLE
Add header site search

### PR DESCRIPTION
**Pull request by grahambancroft**
# Done
Add header search.
Use asset server for assets in header and search SASS files.

## QA
Run gulp build, head over to the demo. Check the search box in large, medium and small screens.
Also check that the search icon on small changes colour if $brand-color is below a certain lightness. 

## Details
Card: https://canonical.leankit.com/Boards/View/111185042/117138508

Fixes: #232

### CODE REVIEW

File: scss/modules/_header.scss [hunk 1]
```diff
@@ -207,26 +207,23 @@
...
-          }
+        &.close {
+          display: block;
         }
       }
     }
   }
-  // scss-lint:enable NestingDepth
-} // @mixin vf-header
+  // scss-lint:enable NestingDepth, IdSelector
```
Comment (anthonydillon) [inline scss/modules/_header.scss]: What ID are you targeting here?

File: scss/modules/_search.scss [hunk 2]
```diff
@@ -84,12 +92,41 @@
...
+        display: none;
+      }
+    }
   }
-  
-  .search-toggle:link,
-  .search-toggle:active {
-    outline: none;
+
+  // scss-lint:disable NestingDepth, IdSelector
```
Comment (anthonydillon) [inline scss/modules/_search.scss]: Again it doesnt look like your targeting any ids

**Comment by anthonydillon** [comment]
The mobile search field seems a little to tight:
vanilla framework pattern library

File: scss/modules/_header.scss (hunk 1 above)
Comment (grahambancroft) [inline scss/modules/_header.scss]: @anthonydillon this targets #nav:hover at line 210

File: scss/modules/_search.scss (hunk 2 above)
Comment (grahambancroft) [inline scss/modules/_search.scss]: @anthonydillon this targets #site-search:hover at line 111

**Comment by grahambancroft** [comment]
@anthonydillon The search field is the same size as live and Insights afaict, I can get an issue and card in for Yaili to take a look?

**Comment by anthonydillon** [comment]
Yeah, if you could show Yaili and see what she thinks please.

**Comment by anthonydillon** [comment]
What is the status of this PR?

**Comment by grahambancroft** [comment]
Yaili has approved the design. This is ready for re-review.

**Comment by anthonydillon** [comment]
LGTM :+1:

**Pull request merged**